From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Mon, 17 Oct 2022 11:35:36 +0200
Subject: [PATCH 010/192] backports: drop QRTR and MHI

Backports currently include QRTR and MHI due to ath11k-pci requiring them,
however this at the same time prevents us from adding ath11k-ahb as it
also requires QRTR however its AHB variant from the kernel will conflict
with the core provided by backports.

Since MHI also conflicts with existing OpenWrt kmods providing MHI drop
both from backports and use the ones provided by OpenWrt kernel.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 Kconfig.sources                         | 2 --
 Makefile.kernel                         | 2 --
 drivers/net/wireless/ath/ath11k/Kconfig | 6 +++---
 local-symbols                           | 8 --------
 4 files changed, 3 insertions(+), 15 deletions(-)

diff --git a/Kconfig.sources b/Kconfig.sources
index 0aa62a3..d74affd 100644
--- a/Kconfig.sources
+++ b/Kconfig.sources
@@ -4,8 +4,6 @@ source "$BACKPORT_DIR/compat/Kconfig"
 # these are copied from the kernel
 source "$BACKPORT_DIR/net/wireless/Kconfig"
 source "$BACKPORT_DIR/net/mac80211/Kconfig"
-source "$BACKPORT_DIR/net/qrtr/Kconfig"
-source "$BACKPORT_DIR/drivers/bus/mhi/Kconfig"
 source "$BACKPORT_DIR/drivers/soc/qcom/Kconfig"
 source "$BACKPORT_DIR/drivers/net/wireless/Kconfig"
 source "$BACKPORT_DIR/drivers/net/usb/Kconfig"
diff --git a/Makefile.kernel b/Makefile.kernel
index 8036c41..5719470 100644
--- a/Makefile.kernel
+++ b/Makefile.kernel
@@ -43,9 +43,7 @@ obj-y += compat/
 
 obj-$(CPTCFG_CFG80211) += net/wireless/
 obj-$(CPTCFG_MAC80211) += net/mac80211/
-obj-$(CPTCFG_QRTR) += net/qrtr/
 obj-$(CPTCFG_QCOM_QMI_HELPERS) += drivers/soc/qcom/
-obj-$(CPTCFG_MHI_BUS) += drivers/bus/mhi/
 obj-$(CPTCFG_WLAN) += drivers/net/wireless/
 obj-$(CPTCFG_USB_NET_RNDIS_WLAN) += drivers/net/usb/
 
diff --git a/drivers/net/wireless/ath/ath11k/Kconfig b/drivers/net/wireless/ath/ath11k/Kconfig
index ec7829d..47cd636 100644
--- a/drivers/net/wireless/ath/ath11k/Kconfig
+++ b/drivers/net/wireless/ath/ath11k/Kconfig
@@ -24,9 +24,9 @@ config ATH11K_PCI
 	tristate "Atheros ath11k PCI support"
 	depends on m
 	depends on ATH11K && PCI
-	select MHI_BUS
-	select QRTR
-	select QRTR_MHI
+	depends on MHI_BUS
+	depends on QRTR
+	depends on QRTR_MHI
 	select PCI_PWRCTRL_PWRSEQ if HAVE_PWRCTRL
 	help
 	  This module adds support for PCIE bus
diff --git a/local-symbols b/local-symbols
index 684dea1..f119ccc 100644
--- a/local-symbols
+++ b/local-symbols
@@ -55,14 +55,6 @@ MAC80211_MESH_PS_DEBUG=
 MAC80211_TDLS_DEBUG=
 MAC80211_DEBUG_COUNTERS=
 MAC80211_STA_HASH_MAX_SIZE=
-QRTR=
-QRTR_SMD=
-QRTR_TUN=
-QRTR_MHI=
-MHI_BUS=
-MHI_BUS_DEBUG=
-MHI_BUS_PCI_GENERIC=
-MHI_BUS_EP=
 QCOM_AOSS_QMP=
 QCOM_COMMAND_DB=
 QCOM_GENI_SE=
